docs/migrating.md

```diff
@@ -0,0 +1,116 @@
+see the [LLVM documentation](https://clang.llvm.org/docs/index.html)
+and [samples](https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/tree/main/samples/src).
+
+LLVM Embedded Toolchain provides superior performance when targeting the
+Armv8-M or later architecture, including the
+[Arm Helium Technology](https://www.arm.com/technologies/helium)
+(M-Profile Vector Extension, MVE).
+
+## Key toolchain components
+
+|Component|GNU toolchain (`arm-none-eabi-`)|LLVM toolchain|
+|---------|-------------|--------------|
+|C, C++ compiler​|`gcc`, `g++`|`clang`, `clang++`​|​
+|Assembler​|`as`​|`clang` integrated assembler​|
+|Linker​|`ld`​|`lld`​|
+|Binutils​|`objdump`, `readelf`, ...|`llvm-objdump`, `llvm-readelf`, ...|
+|Compiler runtime library​|`libgcc​`|`compiler-rt`​|
+|Unwinder​|`libgcc`​|`libunwind`​|
+|C standard library​|`newlib`, `newlib-nano`|`picolibc`|​
+|C++ ABI library​|`libsupc++.a`|`libc++abi`​|​
+|C++ standard library​|`libstdc++​`|`libc++`​|
+
+## Toolchain identification
+
+Toolchain version macros:
+
+|Version number|GNU macro|LLVM macro|
+|-------|---------|----------|
+|Major|`__GNUC__`|`__clang_major__`|
+|Minor|`__GNUC_MINOR__`|`__clang_minor__`|
+|Patch level|`__GNUC_PATCHLEVEL__`|`__clang_patchlevel__`|
+
+## C and C++ language extensions
+
+LLVM supports majority of GNU C and C++ extensions as described in
+[Clang Language Extensions](https://clang.llvm.org/docs/LanguageExtensions.html).
+
+The following feature checking macros can be used to test whether a particular
+feature is supported:
+* [`__has_builtin`](https://clang.llvm.org/docs/LanguageExtensions.html#has-builtin)
+* [`__has_feature`](https://clang.llvm.org/docs/LanguageExtensions.html#has-feature-and-has-extension)
+* [`__has_extension`](https://clang.llvm.org/docs/LanguageExtensions.html#has-feature-and-has-extension)
+* [`__has_attribute`](https://clang.llvm.org/docs/LanguageExtensions.html#has-attribute)
+
+## Assembly language
+
+`clang` should be used instead of `arm-none-eabi-as` to compile assembly
+(`.s` and `.S`) files.
+
+There are minor differences in the assembly syntax between GNU and LLVM
+compilers, however most of the time it is possible to write code that is
+accepted by both.
+
+For example, GNU and LLVM compilers differ in:
+* Use of `.N` and `.W` suffixes for Thumb instructions.
+* Use of `S` suffix for flag setting instructions.
+
+## Multilib support
+
+LLVM toolchain provides multilib support similar to the GNU toolchain,
+see _Using the toolchain_ section in the [README](https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/blob/main/README.md),
+however uses different command line options to control selection of semihosting.
+
+|Use case|GNU options|LLVM options|
+|--------|-----------|------------|
+|No semihosting|`--specs=nosys.specs`|`-lcrt0`|
+|Semihosting|`--specs=rdimon.specs`|`-lcrt0-semihost -lsemihost`|
+|Newlib-nano|`--specs=nano.specs`|Not available: `picolibc` is a derivative of `newlib-nano`.
```

Comment:
I'm not sure whether it is a derivative, I think it is closer to a mashup between AVR C (tinystdio) and newlib . On further study I think it is closer to being an "equivalent" of newlib nano.

Reply:
Agreed, replaced.